Javascript/javascript 2/week3/recipe-app.js

```diff
@@ -0,0 +1,335 @@
+    recipeContainer.appendChild(titleElement);
+    recipeContainer.appendChild(ingredientsList);
+    recipeContainer.appendChild(instructionsElement);
+  }
+
+  function displayAddedIngredient(name, amount) {
+    const ingredientList = document.getElementById("ingredientList");
+
+    const ingredientItem = document.createElement("div");
+    ingredientItem.innerText = `${name}: ${amount || "N/A"}`;
+
+    ingredientList.appendChild(ingredientItem);
+  }
+
+  function getIngredientsList() {
+    const ingredientList = document.getElementById("ingredientList");
+    const ingredients = [];
+
+    // Extract added ingredients from the displayed list
+    ingredientList.childNodes.forEach((ingredient) => {
+      const [name, amount] = ingredient.innerText
+        .split(": ")
+        .map((item) => item.trim());
+      ingredients.push({ NAME: name, AMOUNT: amount });
+    });
+
+    return ingredients;
+  }
+
+  // Function to find a recipe by a provided search word
+  window.searchRecipe = function () {
+    const searchWord = document
+      .getElementById("searchWord")
+      .value.toLowerCase();
+
+    const foundRecipes = recipes.filter((recipe) =>
+      recipe.title.toLowerCase().includes(searchWord)
+    );
+
+    // Display found recipes or show a message
+    if (foundRecipes.length > 0) {
+      recipeGrid.innerHTML = "";
+      foundRecipes.forEach((recipe) => {
+        const recipeCard = createRecipeCard(recipe);
+        recipeGrid.appendChild(recipeCard);
+      });
+    } else {
+      recipeGrid.innerHTML = "<p>The recipe is not found.</p>";
+    }
+  };
+
+  // Sort the recipe array by the amount of ingredients
+  window.sortRecipesByIngredients = function () {
+    const sortedRecipes = [...recipes].sort(
+      (recipe1, recipe2) =>
+        recipe1.ingredients.length - recipe2.ingredients.length
+    );
+
+    // Display sorted recipes
+    recipeGrid.innerHTML = "";
+    sortedRecipes.forEach((recipe) => {
+      const recipeCard = createRecipeCard(recipe);
+      recipeGrid.appendChild(recipeCard);
+    });
+  };
+
+  let cookingTimer;
+
+  window.startCookingTimer = function () {
+    const cookingTimeInput = document.getElementById("cookingTime");
+    const cookingTime = parseInt(cookingTimeInput.value, 10);
+
+    if (isNaN(cookingTime) || cookingTime <= 0) {
+      alert("Please enter a valid cooking time.");
+      return;
+    }
+
+    clearInterval(cookingTimer);
+
+    // Set up a new timer
+    cookingTimer = setTimeout(() => {
+      alert("Cooking time is up! Your dish is ready!");
+      clearInterval(); // Clear the cooking time input label
+    }, cookingTime * 60 * 1000); // Convert minutes to milliseconds
+  };
+
+  let timeSpentOnPage = 0;
+
+  function logTimeSpent() {
+    timeSpentOnPage += 5;
+    console.log(`Time spent on the page: ${timeSpentOnPage} seconds`);
+  }
+  const pageTimer = setInterval(logTimeSpent, 5000); // 5000 milliseconds = 5 seconds
```

Comment:
I can see that logTimeSpent function increments the timeSpentOnPage variable by 5 every 5 seconds, but it never resets/clears the timer. This means that the timer will continue to fire indefinitely, potentially causing performance issues or memory leaks. To prevent this, I would suggest to clear the timer after it logs the time spent:

`function logTimeSpent() {
  timeSpentOnPage += 5;
  console.log(`Time spent on the page: ${timeSpentOnPage} seconds`);
  clearInterval(pageTimer);
}`